consumer/src/main/kotlin/streams/procedures/StreamsSinkProcedures.kt

```diff
@@ -27,13 +44,86 @@ class StreamsSinkProcedures {
+        val configuration = streamsEventSinkConfigMapper.convert(config = properties)
 
         val data = readData(topic, config ?: emptyMap(), configuration)
 
         return data.map { StreamResult(mapOf("data" to it)) }.stream()
     }
 
+    @Procedure("streams.sink.start")
+    fun sinkStart(): Stream<StreamsSinkDTO> {
+        checkEnabled()
+        checkLeader()
+        try {
+            streamsEventSink?.start()
+            return sinkStatus()
+        } catch (e: Exception) {
+            log?.error("Cannot start the Sink because of the following exception", e)
+            return Stream.concat(sinkStatus(),
+                    Stream.of(StreamsSinkDTO("exception", ExceptionUtils.getStackTrace(e))))
+        }
+    }
+
+    @Procedure("streams.sink.stop")
+    fun sinkStop(): Stream<StreamsSinkDTO> {
+        checkEnabled()
+        checkLeader()
+        try {
+            streamsEventSink?.stop()
+            return Stream.of(StreamsSinkDTO("stopped", true))
+        } catch (e: Exception) {
+            log?.error("Cannot stopped the Sink because of the following exception", e)
+            return Stream.concat(sinkStatus(),
+                    Stream.of(StreamsSinkDTO("exception", ExceptionUtils.getStackTrace(e))))
+        }
+    }
+
+    @Procedure("streams.sink.restart")
+    fun sinkRestart(): Stream<StreamsSinkDTO> {
+        val stopped = sinkStop().collect(Collectors.toList())
+        val hasError = stopped.stream().anyMatch { it.name == "exception" }
+        if (hasError) {
+            return stopped.stream()
+        }
+        return sinkStart()
+    }
+
+    @Procedure("streams.sink.config")
+    fun sinkConfig(): Stream<StreamsSinkDTO> {
+        checkEnabled()
+        checkLeader()
+        val configMap = JSONUtils.asMap(streamsSinkConfiguration)
+                .filterKeys { it != "topics" && it != "enabled" && it != "proceduresEnabled" }
+                .mapKeys { it.key.toPointCase() }
+                .mapKeys {
+                    when (it.key) {
+                        "error.config" -> "streams.sink.errors"
+                        "procedures.enabled" -> "streams.${it.key}"
+                        else -> if (it.key.startsWith("streams.sink")) it.key else "streams.sink.${it.key}"
+                    }
+                }
+        val topicMap = streamsSinkConfiguration.topics.asMap()
+                .mapKeys { it.key.key }
+        val invalidTopics = mapOf("invalid_topics" to streamsSinkConfiguration.topics.invalid)
+        return (configMap + topicMap + invalidTopics)
+                .entries.stream()
+                .map { StreamsSinkDTO(it.key, it.value) }
+    }
+
+    @Procedure("streams.sink.status")
+    fun sinkStatus(): Stream<StreamsSinkDTO> {
+        checkEnabled()
+        checkLeader()
+        return Stream.of(StreamsSinkDTO("status", streamsEventSink?.status() ?: StreamsPluginStatus.UNKNOWN))
```

Comment:
Failed local testing for me with: 

```
Cannot convert streams.events.StreamsPluginStatus to AnyValue
```

Reply:
When I did a stop and then a start, I got the same thing (but stop worked and returned true)

Reply:
Yes, it odd because the two test passes:

https://github.com/neo4j-contrib/neo4j-streams/blob/e8dbd7982c47e4178b59a52f48b18220016db602/consumer/src/test/kotlin/integrations/kafka/KafkaStreamsSinkProcedures.kt#L276-L311

Let me covert the status to string